data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java

```diff
@@ -176,9 +175,13 @@ public void start(final Promise<Void> startPromise) {
     // Set up the secret watcher
     private void setupSecretWatcher() {
         try {
+            ReciverVertical
+            this.secretWatcher = new SecretWatcher(secretVolume, this::updateServerConfig);
+            new Thread(this.secretWatcher).start();
             File file = new File(secretVolumePath + "/tls.crt");
```

Comment:
You have created
```
this.tlsKeyFile = new File(tlsKeyFilePath);
this.tlsCrtFile = new File(tlsCrtFilePath);
```

So here we don't need to create the file again. You can just directly use that from the class variable you just created.